launcher/ui/pages/instance/ModFolderPage.cpp

```diff
@@ -195,8 +205,14 @@ void ModFolderPage::updateMods()
 
     auto mods_list = m_model->selectedMods(selection);
     bool use_all = mods_list.empty();
-    if (use_all)
+    if (use_all) {
         mods_list = m_model->allMods();
+        for(Mod* mod : mods_list) {
+            if (mod->metadata()->do_updates==false) {
+                mods_list.removeAll(mod);
+            }
+        }
```

Comment:
i believe that, when removing something from a list, the iterator for that list gets invalidated, do this for loop can actually go out of bounds and/or skip some mods. So, this should probably do a copy of the list for iterating, instead of iterating over the same one that gets modified.